server/plugin/plugin.go

```diff
@@ -158,6 +168,36 @@ func (p *MatterpollPlugin) setProfileImage() error {
 	return nil
 }
 
+// checkHTTPConnection checks if a http request to
+func (p *MatterpollPlugin) checkHTTPConnection() error {
+	siteURL := *p.ServerConfig.ServiceSettings.SiteURL
+	url := siteURL + "/plugins/" + manifest.ID + "/"
+
+	for i := 0; i < 3; i++ {
+		backOfTime := time.Duration(i) * 1 * time.Second
+		time.Sleep(backOfTime)
+
+		resp, err := http.Get(url)
```

Comment:
I just noticed.
Here call `http.Get` simply, but if user is in company proxy environment, this might never reach to the host. Mattermost itself has code for proxy like [here](https://github.com/mattermost/mattermost-server/blob/dac7014b48218909c70f5b84e34b1f16f56bb17a/services/httpservice/client.go#L139)

I've encountered problems related to company proxy in may times, it needs huge effort to resolve this completely. Could we implement helper method like `ping` to mattermost-server?